### PR TITLE
chore: refactor Get and Create AccessToken API, add filter option, update permissions

### PR DESCRIPTION
**Pull request by ShreyaLnuHpe**
## Ticket

[DET-10460](https://hpe-aiatscale.atlassian.net/browse/DET-10460)
[DET-10461](https://hpe-aiatscale.atlassian.net/browse/DET-10461)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-10460]: https://hpe-aiatscale.atlassian.net/browse/DET-10460?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-10461]: https://hpe-aiatscale.atlassian.net/browse/DET-10461?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_user.go [hunk 1]
```diff
@@ -766,26 +772,44 @@ func (a *apiServer) GetAllAccessTokens(
...
 
-	if req.Name != "" {
-		var userIDForGivenUsername model.UserID
-		nameFilterExpr := "%" + req.Name + "%"
-		err := db.Bun().NewSelect().
-			Table("users").
-			Column("id").
-			Where("username ILIKE ?", nameFilterExpr).
-			Scan(ctx, &userIDForGivenUsername)
+	// {"showActive":true,"username":"admin","TokenType":"ACCESS_TOKEN"}
```
Comment (corban-beaird) [inline master/internal/api_user.go:775]: do we want to drop this comment?

File: master/internal/api_user_intg_test.go [hunk 2]
```diff
@@ -992,48 +992,74 @@ func TestGetAllAccessTokens(t *testing.T) {
...
+	require.NoError(t, err)
+	filter := fmt.Sprintf(`{"username":"%s"}`, usernameForGivenUserID)
+
+	tokenInfo1, err := api.GetAccessTokens(ctx, &apiv1.GetAccessTokensRequest{
+		Filter: filter,
 	})
 	require.NoError(t, err)
 	require.NotNil(t, tokenInfo1)
-	require.Equal(t, int32(userID1), tokenInfo1.TokenInfo.UserId)
+	// Loop through all the returned user sessions
```
Comment (corban-beaird) [inline master/internal/api_user_intg_test.go:1004]: should this be user session or access token records?

**Review by corban-beaird — APPROVED** [review]
Only a couple nits, overall fantastic work!

File: master/internal/api_user_intg_test.go (hunk 2 above)
Comment (ShreyaLnuHpe) [reply master/internal/api_user_intg_test.go:1004]: Yes, will correct this

**Pull request merged**